File: app/src/main/java/org/scp/gymlog/ui/main/MainActivity.java
package org.scp.gymlog.ui.main;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import org.scp.gymlog.R;
import org.scp.gymlog.SplashActivity;
import org.scp.gymlog.databinding.ActivityMainBinding;
import org.scp.gymlog.util.Data;

public class MainActivity extends AppCompatActivity {

	private ActivityMainBinding binding;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		if (Data.getInstance().getMuscles().isEmpty()) {
			Intent intent = new Intent(this, SplashActivity.class);
			startActivity(intent);
			finish();
			return;
		}

		binding = ActivityMainBinding.inflate(getLayoutInflater());
		setContentView(binding.getRoot());

		BottomNavigationView navView = findViewById(R.id.navView);
		// Passing each menu ID as a set of Ids because each
		// menu should be considered as top level destinations.
		AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
				R.id.navigationWorkout,
				R.id.navigationExercises,
				R.id.navigationHistory,
				R.id.navigationPreferences
			).build();
		NavController navController = Navigation.findNavController(this, R.id.navHostFragmentActivityMain);
		//NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
		NavigationUI.setupWithNavController(binding.navView, navController);
	}
}

File: app/src/main/java/org/scp/gymlog/ui/main/MainActivity.kt
package org.scp.gymlog.ui.main

import android.content.Intent
import android.os.Bundle
import androidx.navigation.Navigation
import androidx.navigation.ui.NavigationUI
import org.scp.gymlog.R
import org.scp.gymlog.SplashActivity
import org.scp.gymlog.databinding.ActivityMainBinding
import org.scp.gymlog.ui.common.CustomAppCompatActivity
import org.scp.gymlog.util.Data
import org.scp.gymlog.util.extensions.RedirectionExts.goToVariation

class MainActivity : CustomAppCompatActivity() {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		if (Data.exercises.isEmpty() || intent.extras?.getBoolean("loaded", false) != true) {
			val intent = Intent(this, SplashActivity::class.java)
			startActivity(intent)
			finish()
			return
		}

		val binding = ActivityMainBinding.inflate(layoutInflater)
		setContentView(binding.root)

		val navController = Navigation.findNavController(this, R.id.navHostFragmentActivityMain)
		NavigationUI.setupWithNavController(binding.navView, navController)

		val variationId = intent.extras?.getInt("variationId", -1) ?: -1
		if (variationId >= 0) {
			goToVariation(Data.getVariation(variationId))
		}
	}
}
